akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst

```diff
@@ -0,0 +1,98 @@
+.. _-formFields-:
+
+formFields
+==========
+
+Extracts fields from POST requests generated by HTML forms.
+
+Signature
+---------
+
+::
+
+    def formFields(field: <FieldDef[T]>): Directive1[T]
+    def formFields(fields: <FieldDef[T_i]>*): Directive[T_0 :: ... T_i ... :: HNil]
+    def formFields(fields: <FieldDef[T_0]> :: ... <FieldDef[T_i]> ... :: HNil): Directive[T_0 :: ... T_i ... :: HNil]
+
+The signature shown is simplified and written in pseudo-syntax, the real signature uses magnets. [1]_ The type
+``<FieldDef>`` doesn't really exist but consists of the syntactic variants as shown in the description and the examples.
+
+.. [1] See `The Magnet Pattern`_ for an explanation of magnet-based overloading.
+.. _`The Magnet Pattern`: /blog/2012-12-13-the-magnet-pattern/
+
+Description
+-----------
+
+Form fields can be either extracted as a String or can be converted to another type. The parameter name
+can be supplied either as a String or as a Symbol. Form field extraction can be modified to mark a field
+as required or optional or to filter requests where a form field has a certain value:
+
+``"color"``
+    extract value of field "color" as ``String``
+``"color".?``
+    extract optional value of field "color" as ``Option[String]``
```

Comment:
Formatting here does not look very nice. Perhaps just a unordered list and 
```
"`...` – ... 
```

would look better? Maybe it's just me though.

![formfields_ _akka_documentation](https://cloud.githubusercontent.com/assets/120979/7681226/b0cb82e0-fd6e-11e4-90f4-5a9aca46e7c0.png)